io/io/src/RRawFileUnix.cxx

```diff
@@ -116,22 +116,27 @@ void ROOT::Internal::RRawFileUnix::ReadVImpl(RIOVec *ioVec, unsigned int nReq)
 {
 #ifdef R__HAS_URING
    if (RIoUring::IsAvailable()) {
-      RIoUring ring;
-      std::vector<RIoUring::RReadEvent> reads;
-      reads.reserve(nReq);
-      for (std::size_t i = 0; i < nReq; ++i) {
-         RIoUring::RReadEvent ev;
-         ev.fBuffer = ioVec[i].fBuffer;
-         ev.fOffset = ioVec[i].fOffset;
-         ev.fSize = ioVec[i].fSize;
-         ev.fFileDes = fFileDes;
-         reads.push_back(ev);
+      try {
+         RIoUring ring; // throws std::runtime_error
+         std::vector<RIoUring::RReadEvent> reads;
+         reads.reserve(nReq);
+         for (std::size_t i = 0; i < nReq; ++i) {
+            RIoUring::RReadEvent ev;
+            ev.fBuffer = ioVec[i].fBuffer;
+            ev.fOffset = ioVec[i].fOffset;
+            ev.fSize = ioVec[i].fSize;
+            ev.fFileDes = fFileDes;
+            reads.push_back(ev);
+         }
+         ring.SubmitReadsAndWait(reads.data(), nReq);
+         for (std::size_t i = 0; i < nReq; ++i) {
+            ioVec[i].fOutBytes = reads.at(i).fOutBytes;
+         }
+         return;
       }
-      ring.SubmitReadsAndWait(reads.data(), nReq);
-      for (std::size_t i = 0; i < nReq; ++i) {
-         ioVec[i].fOutBytes = reads.at(i).fOutBytes;
+      catch(const std::runtime_error &e) {
+         Warning("RIoUring", "io_uring is not available\n%s", e.what());
```

Comment:
```suggestion
         Warning("RIoUring", "io_uring is unexpectedly not available\n%s", e.what());
```

Reply:
Applied as part of c8fb490c3014883243467dcadcc406f457a78ef3